src/apps/persistence/tools.clj

```diff
@@ -159,6 +159,13 @@
               [:tools.time_limit_seconds :time_limit_seconds])
       (join tool_types)))
 
+(defn- tool-request-status-subselect
+  []
+  (subselect [:tool_request_status_codes :trsc]
+             (fields [(sqlfn :last :trsc.name) :status])
+             (join [:tool_request_statuses :trs] {:trs.tool_request_status_code_id :trsc.id})
+             (where {:trs.tool_request_id :tool_requests.id})))
```

Comment:
With the `last` aggregation function, it might be a good idea to put an explicit order on this subquery in case the rows aren't returned in chronological order. Since you're using a subquery, you could also avoid using the aggregation function altogether by doing a descending sort on `date_assigned` and limiting the result set to one row.